packages/docsearch-react/src/DocSearchModal.tsx

```diff
@@ -129,6 +130,8 @@ export function DocSearchModal({
     [favoriteSearches, recentSearches, disableUserPersonalization]
   );
 
+  const insightsActive = Boolean(insights);
```

Comment:
We could inline `Boolean(insights)` in `getSources` and get rid of this `useEffect` dep, we already have `insights` in the deps.

Reply:
Oh good callout! Moved 😄